src/ophyd_async/tango/_backend/_tango_transport.py

```diff
@@ -0,0 +1,736 @@
+                raise TypeError(f"{tango_resource} has type [{tr_dtype}] not [{dtype}]")
+
+        if tr_format == AttrDataFormat.SPECTRUM:
+            return {"source": tango_resource, "dtype": "array", "shape": [max_x]}
+        elif tr_format == AttrDataFormat.IMAGE:
+            return {"source": tango_resource, "dtype": "array", "shape": [max_y, max_x]}
+
+    else:
+        if tr_dtype in (Enum, CmdArgType.DevState):
+            if tr_dtype == CmdArgType.DevState:
+                trl_choices = list(DevState.names.keys())
+
+            if datatype:
+                if not issubclass(datatype, (Enum, DevState)):
+                    raise TypeError(
+                        f"{tango_resource} has type Enum not {datatype.__name__}"
+                    )
+                if tr_dtype == Enum and is_attr:
+                    choices = tuple(v.name for v in datatype)
+                    if set(choices) != set(trl_choices):
+                        raise TypeError(
+                            f"{tango_resource} has choices {trl_choices} not {choices}"
+                        )
+            return {
+                "source": tango_resource,
+                "dtype": "string",
+                "shape": [],
+                "choices": trl_choices,
+            }
+        else:
+            if datatype and not issubclass(tr_dtype, datatype):
+                raise TypeError(
+                    f"{tango_resource} has type {tr_dtype.__name__} "
+                    f"not {datatype.__name__}"
+                )
+            return {"source": tango_resource, "dtype": tr_dtype_desc, "shape": []}
+
+
+# --------------------------------------------------------------------
+async def get_tango_trl(
+    full_trl: str, device_proxy: Optional[DeviceProxy]
+) -> TangoProxy:
+    device_trl, trl_name = full_trl.rsplit("/", 1)
+    trl_name = trl_name.lower()
+    device_proxy = device_proxy or await DeviceProxy(device_trl)
+
+    # all attributes can be always accessible with low register
+    all_attrs = [attr_name.lower() for attr_name in device_proxy.get_attribute_list()]
+    if trl_name in all_attrs:
+        return AttributeProxy(device_proxy, trl_name)
+
+    # all commands can be always accessible with low register
+    all_cmds = [cmd_name.lower() for cmd_name in device_proxy.get_command_list()]
+    if trl_name in all_cmds:
+        return CommandProxy(device_proxy, trl_name)
+
+    # If version is below tango 9, then pipes are not supported
+    if device_proxy.info().server_version >= 9:
+        # all pipes can be always accessible with low register
+        all_pipes = [pipe_name.lower() for pipe_name in device_proxy.get_pipe_list()]
+        if trl_name in all_pipes:
+            raise NotImplementedError("Pipes are not supported")
+
+    raise RuntimeError(f"{trl_name} cannot be found in {device_proxy.name()}")
+
+
+# --------------------------------------------------------------------
+class TangoTransport(SignalBackend[T]):
```

Comment:
Please can we call this:
```suggestion
class TangoSignalBackend(SignalBackend[T]):
```

Reply:
I think `Transport` is a leftover from when this was in `ophyd.v2`...